app/models/mets_document.rb

```diff
@@ -26,6 +20,11 @@ def valid_mets?
     true
   end
 
+  def all_images_present?
+    image_present = files.map { |file| File.exist?(Rails.root.join(file[:url])) }
```

Comment:
A good place to use all?()

Reply:
Can you tell me more? Currently the `all_images_present?` method only returns true if *all* the images referenced in the xml file are present in the file system. I was worried that if I named it "images_present?" it would be confusing if it returned false when 90% of the images were present but not all. Would you think `all_referenced_images_present?` would be clearer?

Reply:
I meant you could do `files.all? { |file| File.exist?(Rails.root.join(file[:url])) }`
Instead of mapping to the array of booleans and looking for false.

I think all?() will return on the first false and doesn't need to store any booleans in an array.

Reply:
ah, ok, thanks for the clarification! Sounds like a good improvement, I thought you were referring to the naming of the method, so I'm glad I asked.